pywbem_mock/_mainprovider.py

```diff
@@ -811,152 +790,247 @@ def remove_namespace(self, namespace):
+            * If `True`, all direct and indirect subclasses of the specified
+              class or the top-level classes and all of their direct and
+              indirect subclasses are included in the result.
+            * If `None`, the server defined default is False
+              per :term:`DSP0200` (server-implemented default is `False`).
+
+            Note, the semantics of the `DeepInheritance` parameter in
+            :meth:`~pywbem.WBEMConnection.EnumerateInstances` is different.
+
+          LocalOnly (:class:`py:bool`):
+            Indicates that inherited properties, methods, and qualifiers are to
+            be excluded from the returned classes, as follows.
+
+            * If `False`, inherited elements are not excluded.
+            * If `True`, inherited elements are excluded.
+            * If `None`, this parameter is not passed to the WBEM server, and
+              the server-implemented default `True` to be used per
+              :term:`DSP0200`.
+            TODO: Confirm that our default is TRUE
+
+          IncludeQualifiers (:class:`py:bool`):
+            Indicates that qualifiers are to be included in the returned
+            classes, as follows:
+
+            * If `False`, qualifiers are not included.
+            * If `True`, qualifiers are included.
+            * If `None`, this parameter is not passed to the WBEM server, and
+              causes the server-implemented default to be used. :term:`DSP0200`
+              defines that the server-implemented default is `True`.
+            TODO: Confirm we use True for server defined default
+
+          IncludeClassOrigin (:class:`py:bool`):
+            Indicates that class origin information is to be included on each
+            property and method in the returned classes, as follows:
+
+            * If `False`, class origin information is not included.
+            * If `True`, class origin information is included.
+            * If `None`, this the server defined default is False
+              per :term:`DSP0200` (server-implemented default is `False`).
+            TODO: Confirm we use False
 
         Returns:
 
-            return tuple including list of classes
+            List of :class:`~pywbem.CIMClass` that are the enumerated classes.
 
         Raises:
 
             CIMError: CIM_ERR_INVALID_NAMESPACE: invalid namespace,
             CIMError: CIM_ERR_NOT_FOUND: A class that should be a subclass
             of either the root or "ClassName" parameter was not found in the
-            repository. This is probably a repository build error.
+            datastore. This is probably a repository build error.
             CIMError: CIM_ERR_INVALID_CLASS: class defined by the classname
             parameter does not exist.
         """
-        namespace = self.namespace
-        # Validate namespace and get class_store for this namespace
+
+        self.validate_namespace(namespace)
```

Comment:
`get_class_store()` already validates the namespace.

Reply:
NOTE: I redid this later to remove validate from the get_..._store emthods and explicitly call it when required at the beginning of provider request responder methods.

The one thing we do not have is an explicit test for errors of namespace since that it difficult to do without creating new tests of just the MainProvider without getting from the client.